### DISCUSSION TITLE
Task stuck in running status

### DISCUSSION THREAD

**Discussion by NanyinK** [apache/airflow#17823] [Q&A]
Hi,

I tried to run a dag on the web UI but failed.
It's always showing "running" status.

and when I click on the logs, I get this : 
////////////////////////////////////////////////////////
*** Log file does not exist: /data/app/airflow/logs/dag_test/variable_step1/2021-08-14T16:00:00+00:00/1.log
*** Fetching from: http://:8793/log/dag_test/variable_step1/2021-08-14T16:00:00+00:00/1.log
*** Failed to fetch log file from worker. The request to ':///' is missing either an 'http://'                         or 'https://' protocol.
/////////////////////////////////////////////////////////

but when I run both of the steps locally on the server, it will succeed.

//////////////////////////////////////////////////////
[2021-08-25 15:09:06,214] {taskinstance.py:1204} INFO - Marking task as SUCCESS. dag_id=dag_test, task_id=variable_step1, execution_date=20210825T000000, start_date=20210825T070906, end_date=20210825T070906
///////////////////////////////////////////////////////

Could someone give some clues on troubleshooting this?
Thanks

**Comment by NanyinK** [comment]
My dag file:

////////////////////////////////////////////
```
#
# Licensed to the Apache Software Foundation (ASF) under one
# or more contributor license agreements.  See the NOTICE file
# distributed with this work for additional information
# regarding copyright ownership.  The ASF licenses this file
# to you under the Apache License, Version 2.0 (the
# "License"); you may not use this file except in compliance
# with the License.  You may obtain a copy of the License at
#
#   http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing,
# software distributed under the License is distributed on an
# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
# KIND, either express or implied.  See the License for the
# specific language governing permissions and limitations
# under the License.

"""
### Tutorial Documentation
Documentation that goes along with the Airflow tutorial located
[here](https://airflow.apache.org/tutorial.html)
"""
# [START ]
from datetime import timedelta,datetime

# [START import_module]
# The DAG object; we'll need this to instantiate a DAG
from airflow import DAG
# Operators; we need this to operate!
from airflow.operators.bash_operator import BashOperator
from airflow.utils.dates import days_ago

import pytz
import pendulum
tz = pytz.timezone('Asia/Shanghai')
local_tz = pendulum.timezone("Asia/Shanghai")
#local_tz = local_dt.astimezone(pytz.utc)
#local_dt = local_dt.astimezone(pytz.utc)
#dt = datetime(2018, 6, 13, 17, 40, tzinfo=tz)
#utc_dt = dt.astimezone(pytz.utc).replace(tzinfo=None)



# [END import_module]

# [START default_args]
# These args will get passed on to each operator
# You can override them on a per-task basis during operator initialization
default_args = {
    'owner': 'airflow',
    'start_date': datetime(2021, 8, 15, tzinfo=local_tz),
#    'retries': 1,
#    'retry_delay': timedelta(minutes=5),
    # 'queue': 'bash_queue',
    # 'pool': 'backfill',
    # 'priority_weight': 10,
    # 'end_date': datetime(2016, 1, 1),
    # 'wait_for_downstream': False,
    # 'dag': dag,
    # 'sla': timedelta(hours=2),
    # 'execution_timeout': timedelta(seconds=300),
    # 'on_failure_callback': some_function,
    # 'on_success_callback': some_other_function,
    # 'on_retry_callback': another_function,
    # 'sla_miss_callback': yet_another_function,
    # 'trigger_rule': 'all_success'
}
# [END default_args]

# [START instantiate_dag]
with DAG(
    'dag_test',
    default_args=default_args,
    description='clickhouse ETL jobs',
#    schedule_interval=timedelta(days=1),
#    schedule_interval='35 14 * * *',
    schedule_interval='@once',
    tags=['order'],
) as dag:
# [END instantiate_dag]

# t1, t2 and t3 are examples of tasks created by instantiating operators
    # [START basic_task]
    t1 = BashOperator(
        task_id='variable_step1',
        bash_command='touch /tmp/1.txt',
        dag=dag,
    )
    
    t2 = BashOperator(
        task_id='variable_step2',
        depends_on_past=False,
        bash_command='touch /tmp/2.txt',
        retries=3,
        dag=dag,
    )
    # [END basic_task]
    
    # [START documentation]
    dag.doc_md = __doc__
    
    dag.doc_md = """\
    #### Task Documentation
    """
    # [END documentation]
    
    t1 >> [t2]
    # [END ]

```

**Comment by ephraimbuddy** [comment]
If you are not in airflow 2.1.3, can you upgrade and try it on 2.1.3? How did you run the dag on the UI? Unpaused and triggered?

**Comment by NanyinK** [comment]
Hi @ephraimbuddy ,

Thanks for looking into this.

1. Im afraid we can't upgrade to 2.1.3
2. I used 3 methods to run the dag .
 a. unpaused on the UI
 b. click on the "trigger " on the UI
 c. use trigger command on the server.

all of them failed and show me the recent tasks are "queued"

**Comment by ephraimbuddy** [comment]
@NanyinK Just a little more questions:
1. What is your airflow version?
2. What executor are you using?
3. Do you have the scheduler logs when this happens? if yes, please share

**Comment by NanyinK** [comment]
@ephraimbuddy 

1. v2.1.2
2. celery
3. yes the log is saying this : 
////////////////////////////////
[2021-08-25 16:11:13,961] {scheduler_job.py:642} INFO - DAG(s) dict_keys(['dag_test']) retrieved from /data/app/airflow/dag/dag_test.py
[2021-08-25 16:11:13,984] {logging_mixin.py:104} INFO - [2021-08-25 16:11:13,984] {dag.py:1833} INFO - Sync 1 DAGs
[2021-08-25 16:11:13,994] {logging_mixin.py:104} INFO - [2021-08-25 16:11:13,994] {dag.py:2306} INFO - Setting next_dagrun for dag_test to None
[2021-08-25 16:11:14,012] {scheduler_job.py:189} INFO - Processing /data/app/airflow/dag/dag_test.py took 0.079 seconds
[2021-08-25 16:11:44,058] {scheduler_job.py:181} INFO - Started process (PID=17511) to work on /data/app/airflow/dag/dag_test.py
[2021-08-25 16:11:44,059] {scheduler_job.py:632} INFO - Processing file /data/app/airflow/dag/dag_test.py for tasks to queue
[2021-08-25 16:11:44,060] {logging_mixin.py:104} INFO - [2021-08-25 16:11:44,060] {dagbag.py:496} INFO - Filling up the DagBag from /data/app/airflow/dag/dag_test.py

**Comment by NanyinK** [comment]
I found a very strange thing, the job will be queued instead of running once I run it.

/////////////////////////////////////////////////////////
airflow tasks run dag_test variable_step1 2021-08-25T09:46:36.104510+00:00 --local --pool default_pool --subdir /data/app/airflow/dag/dag_test.py
[2021-08-25 17:54:35,614] {dagbag.py:496} INFO - Filling up the DagBag from /data/app/airflow/dag/dag_test.py
Running <TaskInstance: dag_test.variable_step1 2021-08-25T09:46:36.104510+00:00 [queued]> on host 127.0.0.1
//////////////////////////////////////////////////////////

**Comment by NanyinK** [comment]
I think the task is checking dependency, but what the real dependency is ..

**Comment by ephraimbuddy** [comment]
Go to logs/scheduler/dagfilename.py.log and get the logs there, maybe there's an issue while parsing the dags
